### DISCUSSION TITLE
Build and development settings in `vercel.json`

### DISCUSSION THREAD

**Discussion by frandiox** [vercel/vercel#6180] [General]
I believe right now these settings (output directory, install command, etc) can only be modified in Vercel dashboard UI.
It would great for template projects if these settings could be changed at the `vercel.json` level.

For example, I have a [template project](https://github.com/frandiox/vitesse-ssr-template) that uses Vite and pnpm, and deploys an SSR app to a Vercel function. The problem is that Vite already requires a `public` directory which is different from what we need to deploy to Vercel, so we need to move the Vercel root to a subfolder (`serverless` in this project). Also, Vercel expects a `package-lock.json` but this projects uses `pnpm` so it would be great to change the settings directly in this template.

Is this something that could be considered?
Thanks.

**Comment by Ethan-Arrowood** [comment]
Hello @frandiox I'm an engineer at Vercel and have been working on just this feature! It hasn't been released just yet, but I'd love to invite you to try it out early. Would you like to schedule sometime next week to try it out? I'm free most of Monday, Tuesday, and Wednesday. If you are interested please send me an email at ethan.arrowood@vercel.com with some options for us to meet. Thank you!
